common/scripted_effects/wc_scheme_effects.txt

```diff
@@ -96,9 +96,14 @@ stasis_trap_success_effect = {
 	#End of Rival check
 	
 	scope:target = {
-		save_scope_value_as = {
-			name = scheme_discovered
-			value = $DISCOVERED$
+		scope:scheme = {
+			if = {
+				limit = { exists = scope:scheme_discovered }
+			}
+			save_scope_value_as = {
+				name = discovered
+				value = yes
+			}
```

Comment:
Why you set it? `scope:scheme_discovered` does the exactly same thing.

Reply:
Because you can't access the scheme_discovered scope in those events, the scheme is already ended at that point.